File: #0881.boats-to-save-people.cpp
class Solution {
public:
    int numRescueBoats(vector<int>& people, int limit) {

        int size = people.size();
        int i = 0;
        int j = size - 1;
        std::sort(people.begin(), people.end());
        int result = 0;
        while (i <= j) {
            result++;
            if (people[i] + people[j] <= limit) {
                i++;
            }
            j--;
        }
        return result;
    }
};

File: #0881.boats-to-save-people.js
/**
 * @param {number[]} people
 * @param {number} limit
 * @return {number}
 */
var numRescueBoats = function(people, limit) {
    let count = 0;
    let left = 0;
    let right = people.length - 1;
    people.sort((left, right) => left - right);
    while (left <= right) {
        if ((people[right] + people[left]) <= limit) {
            left++;
        }
        right--;
        count++;
    }
    return count;
};
